mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1116,6 +1116,18 @@ OpFoldResult arith::MaximumFOp::fold(FoldAdaptor adaptor) {
   if (matchPattern(adaptor.getRhs(), m_NegInfFloat()))
     return getLhs();
 
+  // max(max(a, b), b) -> max(a, b)
+  // max(max(a, b), a) -> max(a, b)
+  if (auto max = getLhs().getDefiningOp<MaximumFOp>())
+    if (getRhs() == max.getRhs() || getRhs() == max.getLhs())
+      return getLhs();
+
+  // max(a, max(a, b)) -> max(a, b)
+  // max(b, max(a, b)) -> max(a, b)
```

Comment:
I wonder if instead of covering both cases here and for other ops, we should instead use the fact that these are commutative and first move the other `min`/`max` operand to the RHS?

Reply:
> An alternative would be to normalize operands inside the fold itself (e.g., always move the nested max to the RHS)

Yes, I suggested this in the other revision: doing this in the trait would align all the commutative operation and then in-turn simplify all the folders (to avoid checking redundant forms)